spec/models/user_spec.rb

```diff
@@ -135,6 +135,38 @@
     end
   end
 
+  describe '#validations' do
+    context 'password' do
+      let(:user) { build :user }
+
+      context 'when oauth is enabled' do
+        it 'saves without any errors even if the password strength is less' do
+          user.password = 'warsdasdf'
+          user.token = 'access token'
+          expect(user.valid?).to be true
+        end
+      end
+
+      context 'with valid one' do
+        it 'saves without any errors' do
+          user.password = 'waspAr$0'
```

Comment:
Should you also test the case where user.password is blank because of `return if password.blank? || (password =~ PWD_REGEX)`